### DISCUSSION TITLE
Prisma like Return types  🚀

### DISCUSSION THREAD

**Discussion by phifa** [keystonejs/keystone#7438] [Feature requests]
We need something like this: 

<img width="775" alt="Bildschirmfoto 2022-04-06 um 09 26 43" src="https://user-images.githubusercontent.com/3061129/161930703-382d22c4-bacb-4d60-9a1a-1eb69ec796a8.png">

<img width="779" alt="Bildschirmfoto 2022-04-06 um 09 26 55" src="https://user-images.githubusercontent.com/3061129/161930744-adfec286-482d-4e4f-8611-dd2e8e88872c.png">

https://www.prisma.io/blog/mongodb-general-availability-pixnun6mffmu

When working with the `context.query` or the `context.prisma` api, having the returne values typed would be amazing. The question is: Since this is in prisma, will this be automatically be available in the `context.prisma` api? Is there a way to make this available in the `context.query` api? I can see how this could be difficult with the graphql string.

**Comment by dcousens** [comment]
I am replacing this discussion with https://github.com/keystonejs/keystone/discussions/8498, please @phifa if you have any feedback, leave feedback on the proposal we have put forward :yellow_heart:
